src/pattern-matcher.mjs

```diff
@@ -74,10 +72,11 @@ function replaceS(matcher, str, replacement) {
  * Replace a given string by a given matcher.
  * @param {PatternMatcher} matcher The pattern matcher.
  * @param {string} str The string to be replaced.
- * @param {(...strs[])=>string} replace The function to replace each matched part.
+ * @param {(...strs: (string|number)[])=>string} replace The function to replace each matched part.
```

Comment:
How far is this from being merged? Would it be possible to cherry pick this change explicitly? 

Context: 
I'm introducing an Eslint plugin to our workflow that is relying on this package. Our typechecking currently depends on `maxNodeModuleJsDepth` since we have a monorepo including old js-with-types-in-jsdoc modules, resulting in this package being blocked in CI because of invalid syntax.

Reply:
As a person doing lots of js-with-types and have used `maxNodeModuleJsDepth`, I would say: There's never an excuse to use `maxNodeModuleJsDepth` anymore, since one can now compile type declarations from JSDoc declarations.

In a monorepo – either have TypeScript handle the entire monorepo as a single entity (thus allowing JSDoc in all parts) and handle individual module paths through [`paths` in `tsconfig.json`](https://www.typescriptlang.org/tsconfig/#paths) or use something like [`@monorepo-utils/workspaces-to-typescript-project-references`](https://snyk.io/advisor/npm-package/@monorepo-utils/workspaces-to-typescript-project-references) to set up [project references](https://www.typescriptlang.org/docs/handbook/project-references.html) to compile type declarations for the different parts (haven't tried this specific approach with JS)

Published JSDoc comments are not meant for consumption by TypeScript – only published type declarations are

Reply:
> How far is this from being merged?

On this topic, see https://github.com/eslint-community/eslint-utils/issues/232 :)

Reply:
@GNRSN Feel free to pick my brain on the types-in-js approach over at https://github.com/voxpelli/types-in-js :)